consensus/service/src/bin/main.rs

```diff
@@ -67,6 +67,10 @@ fn main() -> Result<(), ConsensusServiceError> {
         &config.peer_responder_id,
         &config.client_responder_id,
         &cached_key,
+        // Note/TODO: Right now the fee map is optionally provided by the tokens configuration
+        // file, and that is the only configurtable parameter in that file. Once the configuration
+        // is extended, we will likely need to pass parts of it (or all of it) to the enclave in
+        // order to include it in the responder id.
```

Comment:
As mentioned in the comment, right now if more configuration is added to `TokenConfig` (e.g. minter keys), it will not make its way into the enclave and not get included in the responder id. This will allow for a configuration mismatch between nodes, which is undesirable. I opted to not address this at this PR, since I am not yet sure about what I want to pass to the enclave when it comes to the new minting configuration.

There are two ways to go around this and I haven't made up my mind on which one is superior since the minting stuff is not written yet.

One approach is to move `TokensConfig` into `mc-consensus-enclave-api` (where `FeeMap` lives), and pass that to the enclave instead of `FeeMap`. This has the advantage of giving the entire configuration to the enclave to do with it as it pleases (such as Digest the whole thing into the `ResponderId`), but means more parsing code in the enclave: continuing with the minting keys example, we're likely going to have something like this in the TOML file:
```
{
    "token_id": 1,
    "master_minter_keys": [
        "... hex encoded Ed25519 key ...",
         ...
     ]
}
```

The hex encoded key then needs to get decoded into `Ed25519Public`, which is a thing we can do in the enclave but maybe we don't want to. It also means that if we want to add any kind of token-specific configuration we need a new enclave version, which may or may not be desirable.

Alternatively, we can synthesize the minting configuration (or whatever other configuration data we add) into a struct that contains parsed and validated data (e.g. `Ed25519Public` fields) and pass that to the enclave.

Personally I am leaning towards the first option since I think the added code footprint in the enclave will be negligible and will overall result in a more pleasant developer experience since there will be less shuffling of data between various structs. It also means the `enclave_init` API call does not need to keep accumulating arguments for token-related configuration.

I'd like to hear if anyone else has any thoughts on this.

Reply:
I'd love to have a separate object that's been completely decoded be the one getting passed into the enclave, if only to avoid TOML/JSON parsing / `std` pollution.

Reply:
That might be where we are converging on - in https://github.com/mobilecoinfoundation/mobilecoin/pull/1468 there's an argument added to pass the BlockVersion to `enclave_init`. It occurred to me that we should create a configuration structure where all of this stuff lives in (block version + fee map, followed by block version + fee map + minting config) that is passed instead of piling more and more arguments to that method.

But, the enclave will not be doing TOML/JSON parsing. That stuff would live behind a feature flag or in a different crate. Separately, I am currently experimenting with some `serde` attributes that might actually require no parsing in the enclave of anything that has to do with the token configuration.